src/main/java/com/redhat/devtools/lsp4ij/features/foldingRange/LSPFoldingRange.java

```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.foldingRange;
+
+import org.eclipse.lsp4j.FoldingRange;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Augments {@link FoldingRange} with add a flag denoting whether or not it should be collapsed by default.
+ */
+class LSPFoldingRange extends FoldingRange {
```

Comment:
Pretty much what it says, but this allows us to capture whether or not the folding range is collapsed by default while building the list of folding ranges from each language server. Note that it's package-protected visibility to prevent API contamination.